In [1]:
#Cargamos el mismo dataset de la google store
import numpy as np
import pandas as pd
import locale

url = "https://raw.githubusercontent.com/mricardo89/data-mining/main/Unit-II/datasets/googleplaystore.csv"
df = pd.read_csv(url)

In [2]:
#RETO 1 -> La trampa de las unidades de medida

def convertToNum(numStr : str) -> float:

    if numStr.endswith("M"): #Represent megabytes
        numStr = numStr[:-1]
        return float(numStr)
    elif numStr.endswith("k"):
        numStr = numStr[:-1]
        newVal = float(numStr) / 1024 #Convert to megabytes
        return newVal

    return np.nan

df['Size'] = df['Size'].apply(convertToNum)
print(df['Size'].mean())


21.51616543577433


 Una vez convertida la columna a valores numéricos (Megabytes), ejecuta el método .mean(). ¿Cuál es el peso promedio en Megabytes de las apps en la Play Store? El promedio es de 21.516 megabytes 

In [ ]:
#RETO 2 ->El Tipo de Dato Cronológico
locale.setlocale(locale.LC_TIME, 'en_US.UTF-8') #Convert to english to use correctly the format to not have rpoblems with the spanish fro the months
df['Last Updated'] = pd.to_datetime(df['Last Updated'], format='%B %d, %Y', errors='coerce') #Convert to specific datetime using this format

df['Year_Updated'] = df['Last Updated'].dt.year
print("Año con mayor cantidad de aplicaciones : ", df['Year_Updated'].value_counts())


Año con mayor cantidad de aplciaciones :  Year_Updated
2018.0    7349
2017.0    1867
2016.0     804
2015.0     459
2014.0     209
2013.0     110
2012.0      26
2011.0      15
2010.0       1
Name: count, dtype: int64


 Utilizando la sumarización categórica (value_counts()) sobre tu nueva columna Year_Updated, ¿en qué año se actualizó la mayor cantidad de aplicaciones en nuestro dataset? El año en donde se actualizo la mayor cantidad de aplicaciones fue en el 2018 con 7,349

In [12]:
#RETO C -> La desicion arquitectonica
df['Size'] = df['Size'].fillna(df['Size'].median())
print("Cant of Nan :", df['Size'].isna().sum())

Cant of Nan : 0


 Redacten una breve justificación técnica de 3 líneas explicando qué método eligieron y por qué lo consideran superior para no dañar al modelo de predicción.

Despues de analizar el problema, pude determinar que lo mejor para estasituacion es rellenar el hueco utilizando la mediana global, ayq eu el objetivo de nuestro modelo de prediccion es saber si las aplicaciones mas pesadas tienen menos descargas, por lo que el target seria descargas, y es necesario tener los valores de Size para evitar un sesgo, ademas de que solo hay 9145 objetos non-null, haciendo un 10% de posibles datos que son importantes para la prediccion.